src/sites/dMCDA.jl

```diff
@@ -182,12 +205,47 @@ function rank_sites!(S, weights, rankings, n_site_int, mcda_func, rank_col)::Tup
     return prefsites, s_order
 end
 
-function rank_seed_sites!(S, weights, rankings, n_site_int, mcda_func)::Tuple{Vector{Int64},Matrix{Union{Float64,Int64}}}
-    rank_sites!(S, weights, rankings, n_site_int, mcda_func, 2)
+"""
+    retrieve_ranks(S::Matrix, weights::Vector{Float64}, mcda_func::Function, site_ids::Vector)
+    retrieve_ranks(S::Matrix, weights::Vector{Float64}, mcda_func::Vector{Any}, site_ids::Vector)
+    retrieve_ranks(S::Matrix, scores::Vector, rev_val::Bool, site_ids::Vector)
+
+Get location ranks using mcda technique specified in mcda_func, weights and a decision matrix S.
+
+# Arguments
+- `S` : decision matrix containing criteria values for each location (n locations)*(m criteria)
+- `weights` : importance weights for each criteria. 
+- `mcda_func` : function/[function bool] array to use for mcda, specified as an element from method.
+- `site_ids` : array of site ids still remaining after filtering.
+- `scores` : set of scores derived from applying an mcda ranking method.
+- `rev_val` : Boolean indicating whether a mcda method is maximising score (true), or minimising (false). 
+
+# Returns
+- `s_order` : [site_ids, criteria values, ranks]
+"""
+function retrieve_ranks(S::Matrix, weights::Vector{Float64}, mcda_func::Function, site_ids::Vector)
+    S = mcda_normalize(S)
+    S .= S .* repeat(weights', size(S, 1), 1)
+    scores = mcda_func(S)
+
+    return retrieve_ranks(S, vec(scores), true, site_ids)
 end
-function rank_shade_sites!(S, weights, rankings, n_site_int, mcda_func)::Tuple{Vector{Int64},Matrix{Union{Float64,Int64}}}
-    rank_sites!(S, weights, rankings, n_site_int, mcda_func, 3)
+function retrieve_ranks(S::Matrix, weights::Vector{Float64}, mcda_func::Vector{Any}, site_ids::Vector)
+    fns = repeat([maximum], length(weights))
+    results = mcdm(MCDMSetting(S, weights, fns), mcda_func[1])
+
+    return retrieve_ranks(S, results.scores, mcda_func[2], site_ids)
 end
+function retrieve_ranks(S::Matrix, scores::Vector, rev_val::Bool, site_ids::Vector)
+
+    s_order = Union{Float64,Int64}[Int.(site_ids) scores 1:size(S, 1)]
```

Comment:
Use concrete types where possible please: `Int` ->`Int64`

And you can consolidate this line with Line 243 to avoid reallocations.